sui_types/src/serialize.rs

```diff
@@ -105,14 +105,6 @@ where
     serialize_into(writer, &ShallowSerializedMessage::Cert(value))
```

Comment:
The rest of the visitor's elements only seem to be used in unit tests, is it worth ripping off the whole band-aid?

Reply:
Yes my plan is to completely rip the rest of this out, I just had people bugging me about getting some of the changes. in the PR out asap. I'm headed back to so if i can finish ripping this out. Mostly the biggest headache this should solve is the transferring of opaque Bytes around the place and instead force us to better use type saftey (as you can see with the Benchmarking changes)